docs/safe_lambda_deployments.rst

```diff
@@ -180,20 +189,39 @@ Hooks are extremely powerful because:
   you can directly invoke the version to run integration tests or even pre-warm the Lambda containers before exposing
   to production traffic.
 
+    NOTE: The event payload delivered to the Hook function will not contain the Lambda ARN to be tested.
+    We recommend adding an Envrionment variable to the Hook function that maintains the current version of the function requiring safe deployments
+
+.. code:: yaml
+
+  CurrentVersion: !Ref MySafeLambdaFunction.Version
+
+
 - **Hooks are executed per-function**: Each Lambda function gets its own PreTraffic and PostTraffic hook (technically
   speaking hooks are executed once per DeploymentGroup, but in this case the DeploymentGroup contains only one Lambda
   Function). So you can customize the hooks logic to the function that is being deployed.
 
-Checkout the examples folder for an example for how to create a hook function.
-
     NOTE: If the Hook functions are created by the same SAM template that is deployed, then make sure to turn off
-    traffic shifting deployments for the hook functions. 
+    traffic shifting deployments for the hook functions. Also, you should not rely on SAM to create the default function Role, since it
+    will not contain the necessary permissions to call the CodeDepoloy APIs or Invoke your new Lambda function for testing.
+    Instead, either use the Role_ property to specify your own custom Role or use the Policies_ attribute to provide the permissions needed. Finally, use the ``FunctionName`` property to control the exact name of the Lambda function Cloudformation creates. Otherwise, Cloudformation will create your Lambda function with the Stack name and a unique ID added as part of the name.
+
+.. _Role: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction
+.. _Policies: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#resource-types
+
+.. code:: yaml
+
+    FunctionName: 'CodeDeployHook_preTrafficHook'
+    DeploymentPreference:
+        Enabled: false
+    Role: [your Custom RoleArn Here]
+    Policies:
+     - [YourPolicyTemplateHere]
```

Comment:
Add the correct Policy that is needed here

Reply:
Done. Policies added.